src/SDK/SmartDetectorsShared/Clients/AnalysisServicesFactory.cs

```diff
@@ -7,28 +7,25 @@
 namespace Microsoft.Azure.Monitoring.SmartDetectors.Clients
 {
     using System;
-    using System.Collections.Concurrent;
-    using System.Collections.Generic;
+    using System.Configuration;
     using System.Globalization;
-    using System.Linq;
     using System.Threading;
     using System.Threading.Tasks;
     using Microsoft.Azure.Monitoring.SmartDetectors;
     using Microsoft.Azure.Monitoring.SmartDetectors.ActivityLog;
     using Microsoft.Azure.Monitoring.SmartDetectors.Arm;
     using Microsoft.Azure.Monitoring.SmartDetectors.Metric;
-    using Newtonsoft.Json;
-    using Newtonsoft.Json.Linq;
     using ResourceType = Microsoft.Azure.Monitoring.SmartDetectors.ResourceType;
 
     /// <summary>
     /// An implementation of the <see cref="IAnalysisServicesFactory"/> interface.
     /// </summary>
     public class AnalysisServicesFactory : IInternalAnalysisServicesFactory
     {
-        private const int MaxNumberOfResourcesInQuery = 10;
-        private readonly ConcurrentDictionary<string, IList<ResourceIdentifier>> subscriptionIdToWorkspaces = new ConcurrentDictionary<string, IList<ResourceIdentifier>>(StringComparer.CurrentCultureIgnoreCase);
-        private readonly ConcurrentDictionary<string, ResourceIdentifier> aksClusterIdToWorkspaces = new ConcurrentDictionary<string, ResourceIdentifier>(StringComparer.CurrentCultureIgnoreCase);
+        private const string ApplicationInsightsQueryUriFormat = "https://api.applicationinsights.io/v1/apps/{0}/query";
+        private const string LogAnalyticsQueryUriFormat = "https://api.loganalytics.io/v1/workspaces/{0}/query";
+        private const string ResourceCentricQueryUriFormat = "https://api.loganalytics.io/v1{0}/query";
```

Comment:
why these are defined here and in config? Isnt config not enough?
Also do you need sometimes to go to int environment. For tests for example?

Reply:
talked it over - this is only in the emulator's configuration

---
In reply to: [301941642](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/68#discussion_r301941642) [](ancestors = 301941642)